src/main/java/jenkins/plugins/rpmsign/Rpm.java

```diff
@@ -4,15 +4,24 @@
 import hudson.model.AbstractDescribableImpl;
 import hudson.model.Descriptor;
 import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
 
 public class Rpm extends AbstractDescribableImpl<Rpm> {
 
-  private final String gpgKeyName;
-  private final String includes;
-  private final String cmdlineOpts;
-  private final boolean resign;
+  private String gpgKeyName;
+  private String includes;
+  private String cmdlineOpts;
+  private boolean resign;
 
   @DataBoundConstructor
+  public Rpm() {
+    this.gpgKeyName = "";
+    this.includes = "**/**.rpm";
+    this.resign = false;
+    this.cmdlineOpts = "";
+  }
+
+  @Deprecated
   public Rpm(String gpgKeyName, String includes, String cmdlineOpts, boolean resign) {
```

Comment:
Why deprecate the old constructor? Do pipeline uses of this plugin use the setters instead of the constructor? Wouldn't this constructor still be appropriate for non-pipeline configurations?

Reply:
pipeline uses constructor, and setter annotated with @DataBoundConstructor,   @DataBoundSetter annotations respectively.
that's a valid point. We'll have to check if it's required for non-pipeline configurations.